File: app/src/main/java/education/mahmoud/quranyapp/data_layer/local/room/BookmarkDAO.java
package education.mahmoud.quranyapp.data_layer.local.room;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface BookmarkDAO {

    @Insert
    void addBookmark(BookmarkItem item);

    @Update
    void updateBookmark(BookmarkItem item);

    @Delete
    void delteBookmark(BookmarkItem item);

    @Query("select * from bookmark")
    List<BookmarkItem> getBookmarks();


}

File: app/src/main/java/education/mahmoud/quranyapp/datalayer/local/room/BookmarkDAO.kt
package education.mahmoud.quranyapp.datalayer.local.room

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import io.reactivex.Flowable

@Dao
interface BookmarkDAO {
    @Insert
    fun addBookmark(item: BookmarkItem)

    @Update
    fun updateBookmark(item: BookmarkItem)

    @Delete
    fun deleteBookmark(item: BookmarkItem)

    @Query("select * from bookmark")
    fun loadBookmarks(): Flowable<List<BookmarkItem>>
}